src/HexagonOptimize.cpp

```diff
@@ -634,6 +634,7 @@ class OptimizePatterns : public IRMutator2 {
             // that these patterns are not completely incorrect.
             { "halide.hexagon.satw_add_mpy.vw.vh.h", wild_i32x + wild_i32x*bc(wild_i32), Pattern::ReinterleaveOp0 | Pattern::NarrowOp1 | Pattern::NarrowOp2 },
             { "halide.hexagon.satw_add_mpy.vw.vh.h", wild_i32x + bc(wild_i32)*wild_i32x, Pattern::ReinterleaveOp0 | Pattern::NarrowOp1 | Pattern::NarrowOp2 | Pattern::SwapOps12 },
+            { "halide.hexagon.add_mpy.vw.vh.h", wild_i32x + bc(wild_i32)*wild_i32x, Pattern::ReinterleaveOp0 | Pattern::NarrowOp1 | Pattern::NarrowOp2 | Pattern::SwapOps12 | Pattern::v65orLater },
```

Comment:
I don't think this will ever get selected, because the pattern the line above will match first. I think these should just go ahead of the above two patterns in the list, so they're chosen first (if v65 is enabled). It's also missing the commuted version of the multiply like the above. 

Or, maybe not bother with these because signed integer overflow is undefined anyways, so both of these patterns are technically the same, and maybe it's worse to produce different undefined results on different targets...